prov/rxm/src/rxm_init.c

```diff
@@ -82,21 +85,57 @@ int rxm_info_to_rxm(uint32_t version, struct fi_info *core_info,
 			core_info->tx_attr->iov_limit),
 			core_info->tx_attr->rma_iov_limit);
 
-	*info->rx_attr = *rxm_info.rx_attr;
+	*info->rx_attr = *rxm_info->rx_attr;
 	info->rx_attr->iov_limit = MIN(info->rx_attr->iov_limit,
 			core_info->rx_attr->iov_limit);
 
-	*info->ep_attr = *rxm_info.ep_attr;
+	*info->ep_attr = *rxm_info->ep_attr;
 	info->ep_attr->max_msg_size = core_info->ep_attr->max_msg_size;
 
-	*info->domain_attr = *rxm_info.domain_attr;
+	*info->domain_attr = *rxm_info->domain_attr;
 	info->domain_attr->mr_mode = core_info->domain_attr->mr_mode;
 	info->domain_attr->cq_data_size = MIN(core_info->domain_attr->cq_data_size,
-					      rxm_info.domain_attr->cq_data_size);
+					      rxm_info->domain_attr->cq_data_size);
 
 	return 0;
 }
 
+static int rxm_init_info(void)
+{
+	int ret = 0, param;
+
+	if (rxm_info)
+		return 0;
+
+	fastlock_acquire(&rxm_info_lock);
+	if (rxm_info)
+		goto unlock;
+
+	if (!(rxm_info = fi_dupinfo(&rxm_info_static))) {
```

Comment:
If you don't use rxm_info_static anywhere else, why not just update it directly?